codegen/projections/high_score_service/lib/high_score_service/types.rb

```diff
@@ -9,6 +9,25 @@
 
 module HighScoreService
   module Types
+    # Permitted params for a High Score
+    #
+    # @!attribute game
+    #   The game for the high score
+    #
+    #   @return [String]
+    #
+    # @!attribute score
+    #   The high score for the game
+    #
+    #   @return [Integer]
+    #
+    HighScoreParams = ::Struct.new(
```

Comment:
I think these structures were alphabetized. Would you be able to get them back that way?

Reply:
How important is the alphabetical ordering of the generated structures?

So in order to achieve this as well as adhering to the directed codegen architecture, we can try collecting all the structure/enum/union shapes prior to code generation. E.g.

```
class DirectedRubyCodegen {
    @Override
    public void generateStructure(...) {
        // Collect all structure shapes
    }

    @Override
    public void generateError(...) {
        // Collect all error structure shapes
    }

    @Override
    public void generateUnion(...) {
        // Collect all union shapes
    }

    @Override
    public void customizeBeforeIntegrations(...) {
        // Sort collected shapes alphabetically and generate types.rb file
    }

}
```

Reply:
Whatever works - All shapes, operations, errors, etc, should be alphabetized. It's predictable and easier to debug that way.